AutomatedTesting/Gem/PythonTests/Atom/atom_utils/material_canvas_utils.py

```diff
@@ -5,15 +5,106 @@
 SPDX-License-Identifier: Apache-2.0 OR MIT
 """
 
-import azlmbr.atomtools
+import azlmbr.atomtools as atomtools
+import azlmbr.editor.graph as graph
 import azlmbr.math as math
 import azlmbr.bus as bus
 
+from Atom.atom_utils.atom_constants import (
+    DynamicNodeManagerRequestBusEvents, GraphControllerRequestBusEvents, GraphDocumentRequestBusEvents)
 
-def get_graph_name(document_id: math.Uuid) -> str:
+
+def get_graph_name(graph_document_id: math.Uuid) -> str:
+    """
+    Gets the graph name of the given graph_document_id and returns it as a string.
+    :param graph_document_id: The UUID of a given graph document file.
+    :return: str representing the graph name of graph_document_id
+    """
+    return atomtools.GraphDocumentRequestBus(bus.Event, GraphDocumentRequestBusEvents.GET_GRAPH_NAME, graph_document_id)
+
+
+def get_graph_id(graph_document_id: math.Uuid) -> int:
+    """
+    Gets the graph ID of the given graph_document_id and returns it as an int.
+    :param graph_document_id: The UUID of a given graph document file.
+    :return: int representing the graph ID of graph_document_id
+    """
+    return atomtools.GraphDocumentRequestBus(bus.Event, GraphDocumentRequestBusEvents.GET_GRAPH_ID, graph_document_id)
+
+
+def get_graph(graph_document_id: math.Uuid) -> object:
+    """
+    Gets the graph object of the given graph_document_id and returns it.
+    :param graph_document_id: The UUID of a given graph document file.
+    :return: An AZStd::shared_ptr<Graph> object.
```

Comment:
it appears to be correct in my testing:
```
material_graph = material_canvas_utils.get_graph(material_graph_document_ids[0])
] type(material_graph)
<class 'azlmbr.object.PythonProxyObject'>
] material_graph.typename
AZStd::shared_ptr<Graph>
```
The `AZStd::shared_ptr<Graph>` above is a string but the python terminal lacks the `""` around it.

Reply:
going to reword the docstrings to make this more clear and say `:return: An azlmbr.object.PythonProxyObject representing an AZStd::shared_ptr<Node> object from C++` because we are just getting a proxy for a shared pointer not an actual AZStd object